### ISSUE TITLE
vid2depth - Cannot reproduce depth estimation results with default Hyperparameter Values

### ISSUE THREAD

**Issue by ajithkrishnan** [tensorflow/models#6169]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: * models/research/vid2depth*
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: *Yes* - I have included code to estimate the validation loss
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: *Linux Ubuntu 16.04*
- **TensorFlow installed from (source or binary)**: *via pip*
- **TensorFlow version (use command below)**: *1.8.0*
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**:  *cuda 9.0*
- **GPU model and memory**: *Tesla V100-SXM2*
- **Exact command to reproduce**:     * python inference.py   --kitti_dir kitti-raw-uncompressed   --output_dir inference   --kitti_video kitti-raw-uncompressed/2011_09_26/2011_09_26_drive_0001_sync   --model_ckpt trained-model/model-119496*

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
I followed the instructions given in README.md to analyse the depth estimation performance of the pretrained model (model-119496). I would like to reproduce the results by training the network from scratch. 
I used the entire KITTI dataset (175 GB - mentioned in the README) and i trained the networks using the default Hyperparameter values provided in *train.py*. However my results are subpar. I altered the train_steps from 200,000 (default in train.py) to 119,496 (referring to the filename of the pretrained-model). This too did not produce the expected results. I referred to the paper, the github issues section and stackoverflow, but could not find any extra information regarding this topic. 

I have attached screenshots of the results from inference. I have used the same frame from the kitti-raw-eigen dataset for comparison. Left - The model i trained for 119496 Iterations; Centre - Pretrained model from Tensorflow; Right - Model i trained for 212645 iterations.
As can be seen, in case of both the models i have trained, there are pixels which are estimated to have infinite depth.  

Could you please recommend some directions to optimize the network? Should i further reduce the learning rate form 0.0002 or change the batch size from 4.

### Source code / logs
Here is the train.py i updated to included to estimate the validation loss along with the training loss.

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import math
import os
import random
import time
from absl import app
from absl import flags
from absl import logging
import model
import reader
import numpy as np
import tensorflow as tf
import util

gfile = tf.gfile

HOME_DIR = os.path.expanduser('~')
DEFAULT_DATA_DIR = os.path.join(HOME_DIR, 'vid2depth/data/kitti_raw_eigen')
DEFAULT_CHECKPOINT_DIR = os.path.join(HOME_DIR, 'vid2depth/checkpoints')

flags.DEFINE_string('data_dir', DEFAULT_DATA_DIR, 'Preprocessed data.')
flags.DEFINE_float('learning_rate', 0.0002, 'Adam learning rate.')
flags.DEFINE_float('beta1', 0.9, 'Adam momentum.')
flags.DEFINE_float('reconstr_weight', 0.85, 'Frame reconstruction loss weight.')
flags.DEFINE_float('smooth_weight', 0.05, 'Smoothness loss weight.')
flags.DEFINE_float('ssim_weight', 0.15, 'SSIM loss weight.')
flags.DEFINE_float('icp_weight', 0.0, 'ICP loss weight.')
flags.DEFINE_integer('batch_size', 4, 'The size of a sample batch')
flags.DEFINE_integer('img_height', 128, 'Input frame height.')
flags.DEFINE_integer('img_width', 416, 'Input frame width.')
### Note: Training time grows linearly with sequence length.  Use 2 or 3.
flags.DEFINE_integer('seq_length', 3, 'Number of frames in sequence.')
flags.DEFINE_string('pretrained_ckpt', None, 'Path to checkpoint with '
                    'pretrained weights.  Do not include .data* extension.')
flags.DEFINE_string('checkpoint_dir', DEFAULT_CHECKPOINT_DIR,
                    'Directory to save model checkpoints.')
flags.DEFINE_integer('train_steps', 500000, 'Number of training steps.')
flags.DEFINE_integer('summary_freq', 100, 'Save summaries every N steps.')
flags.DEFINE_bool('legacy_mode', False, 'Whether to limit losses to using only '
                  'the middle frame in sequence as the target frame.')
FLAGS = flags.FLAGS

MAX_TO_KEEP = 500

NUM_SCALES = 4

def main(_):

  seed = 8964
  tf.set_random_seed(seed)
  np.random.seed(seed)
  random.seed(seed)

  if FLAGS.legacy_mode and FLAGS.seq_length < 3:
    raise ValueError('Legacy mode supports sequence length > 2 only.')

  if not gfile.Exists(FLAGS.checkpoint_dir):
    gfile.MakeDirs(FLAGS.checkpoint_dir)

  train_model = model.Model(data_dir=FLAGS.data_dir,
                            is_training=True,
                            learning_rate=FLAGS.learning_rate,
                            beta1=FLAGS.beta1,
                            reconstr_weight=FLAGS.reconstr_weight,
                            smooth_weight=FLAGS.smooth_weight,
                            ssim_weight=FLAGS.ssim_weight,
                            icp_weight=FLAGS.icp_weight,
                            batch_size=FLAGS.batch_size,
                            img_height=FLAGS.img_height,
                            img_width=FLAGS.img_width,
                            seq_length=FLAGS.seq_length,
                            legacy_mode=FLAGS.legacy_mode)

  ### creating an instance of DataReader to obtain the necessary image_stacks
  ### which can then be passed to the session in the feed_dict
  train_model.reader = reader.DataReader(train_model.data_dir, train_model.batch_size,
                                      train_model.img_height, train_model.img_width,
                                      train_model.seq_length, NUM_SCALES)

  ### returns the training image stack
  (image_stack_training, _, _) = (train_model.reader.read_data('train')) 
  ###print(image_stack_train)

  ### Returns the validation image stack
  (image_stack_validation, _, _) = (train_model.reader.read_data('val'))

  train(train_model, FLAGS.pretrained_ckpt, FLAGS.checkpoint_dir,
        FLAGS.train_steps, FLAGS.summary_freq, image_stack_training, image_stack_validation)

def train(train_model, pretrained_ckpt, checkpoint_dir, train_steps,
          summary_freq, image_stack_train, image_stack_val):
  """Train model."""
  best_val_loss = 10
  if pretrained_ckpt is not None:
    vars_to_restore = util.get_vars_to_restore(pretrained_ckpt)
    pretrain_restorer = tf.train.Saver(vars_to_restore)
  vars_to_save = util.get_vars_to_restore()
  saver = tf.train.Saver(vars_to_save + [train_model.global_step],
                         max_to_keep=MAX_TO_KEEP)
  sv = tf.train.Supervisor(logdir=checkpoint_dir, save_summaries_secs=0,
                           saver=None)
  config = tf.ConfigProto()
  config.gpu_options.allow_growth = True
  with sv.managed_session(config=config) as sess:
    if pretrained_ckpt is not None:
      logging.info('Restoring pretrained weights from %s', pretrained_ckpt)
      pretrain_restorer.restore(sess, pretrained_ckpt)
    logging.info('Attempting to resume training from %s...', checkpoint_dir)
    checkpoint = tf.train.latest_checkpoint(checkpoint_dir)
    logging.info('Last checkpoint found: %s', checkpoint)
    if checkpoint:
      saver.restore(sess, checkpoint)
      
    logging.info('Training...')
    start_time = time.time()
    last_summary_time = time.time()
    steps_per_epoch = train_model.reader.steps_per_epoch
    step = 1
    
    while step <= train_steps:
      train_model.image_stack = image_stack_train

      fetches = {
          'train': train_model.train_op,
          'global_step': train_model.global_step,
          'incr_global_step': train_model.incr_global_step
      }     

      if step % summary_freq == 0:
        fetches['loss'] = train_model.total_loss        
        fetches['summary'] = sv.summary_op

      #results = sess.run(fetches, feed_dict={train_model.image_stack: image_stack_train})      
      results = sess.run(fetches)
      
      global_step = results['global_step']

      #print("step = {}".format(step))
      #print('global_step = {}'.format(global_step))
      #print('train_steps = {}'.format(train_steps))

      if step % summary_freq == 0:
        #print('step / summary_freq == 0')
        train_model.image_stack = image_stack_val
        fetches_val = {
          'validation_loss': train_model.total_loss,
        }
        results_val, validation_summary = sess.run([fetches_val, train_model.validation_loss])
        

      if step % summary_freq == 0:        
        sv.summary_writer.add_summary(results['summary'], global_step)
        sv.summary_writer.add_summary(validation_summary, global_step)
        train_epoch = math.ceil(global_step / steps_per_epoch)
        train_step = global_step - (train_epoch - 1) * steps_per_epoch
        this_cycle = time.time() - last_summary_time
        last_summary_time += this_cycle
        logging.info(
            'Epoch: [%2d] [%5d/%5d] iteration no.: %.3f time: %4.2fs (%ds total) loss: %.3f validation_loss: %.3f',
            train_epoch, train_step, steps_per_epoch, global_step, this_cycle,
            time.time() - start_time, results['loss'], results_val['validation_loss'])
        #print('Validation Summary: {}'.format(validation_summary))

      if step % steps_per_epoch == 0:
        ###print('step steps per epoch')
        if results_val['validation_loss'] < best_val_loss:
          print('[*] Validation Loss decreased from {} to {}. Saving checkpoint to {}...'.format(best_val_loss, results_val['validation_loss'], checkpoint_dir))
          best_val_loss = results_val['validation_loss']
          saver.save(sess, os.path.join(checkpoint_dir, 'model'),
                        global_step=global_step)
        else:
            print('[*] Validation loss did not decrease from {}. Saving checkpoint to {}...'.format(best_val_loss, checkpoint_dir))
            saver.save(sess, os.path.join(checkpoint_dir, 'last_model'))

      ### Setting step to global_step allows for training for a total of
      ### train_steps even if the program is restarted during training.
      step = global_step + 1

if __name__ == '__main__':
  app.run(main)
![vid2dpeth_compare](https://user-images.githubusercontent.com/30627046/52472219-0c6a7100-2b93-11e9-8c54-1515e9c73bc9.png)

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
